cmd/podman/machine/machine.go

```diff
@@ -64,3 +77,107 @@ func getMachines(toComplete string) ([]string, cobra.ShellCompDirective) {
 	}
 	return suggestions, cobra.ShellCompDirectiveNoFileComp
 }
+
+func initMachineEvents(cmd *cobra.Command, _ []string) error {
+	logrus.Debugf("Called machine %s.PersistentPreRunE(%s)", cmd.Name(), strings.Join(os.Args, " "))
+
+	sockPaths, err := resolveEventSock()
+	if err != nil {
+		return err
+	}
+
+	// No sockets found, so no need to publish events...
+	if len(sockPaths) == 0 {
+		return nil
+	}
+
+	for _, path := range sockPaths {
+		conn, err := (&net.Dialer{}).DialContext(registry.Context(), "unix", path)
+		if err != nil {
+			logrus.Warnf("Failed to open event socket %q: %v", path, err)
+			continue
+		}
+		logrus.Debugf("Machine event socket %q found", path)
+		sockets = append(sockets, conn)
+	}
+	return nil
+}
+
+func resolveEventSock() ([]string, error) {
+	// Used mostly for testing
+	if sock, found := os.LookupEnv("PODMAN_MACHINE_EVENTS_SOCK"); found {
+		return []string{sock}, nil
+	}
+
+	xdg, err := util.GetRuntimeDir()
+	if err != nil {
+		logrus.Warnf("Failed to get runtime dir, machine events will not be published: %s", err)
+		return nil, nil
+	}
+
+	re := regexp.MustCompile(`machine_events.*\.sock`)
+	sockPaths := make([]string, 0)
+	fn := func(path string, info os.DirEntry, err error) error {
+		switch {
+		case err != nil:
+			return err
+		case info.IsDir():
+			return nil
+		case info.Type() != os.ModeSocket:
+			return nil
+		case !re.MatchString(info.Name()):
+			return nil
+		}
+
+		logrus.Debugf("Machine events will be published on: %q", path)
+		sockPaths = append(sockPaths, path)
+		return nil
+	}
+	if err := filepath.WalkDir(filepath.Join(xdg, "podman"), fn); err != nil {
```

Comment:
this fails because the directory has not been made on a "new" installation.  consider something like:

sDir := filepath.Join(xdg/podman)
os.MakeDirAll(sDir...